nupic/embodied/agents/curious_ppo_agent.py

```diff
@@ -508,76 +525,90 @@ def log_post_update(self, info):
 
         return info
 
-    def auxiliary_loss(self, acs, obs, last_obs):
-        # Update features of the policy network to minibatch obs and acs
-        # TODO: No need to update features if there is no aux task. When there is a task
-        # I think we also don't need to update the policy features, just policy.ac and
-        # policy.ob (at least as long we don't share features with policy) This saves a
-        # forward pass.
-        self.policy.update_features(obs, acs)
-
-        # Update features of the auxiliary network to minibatch obs and acs
-        # Using first element in dynamics list is sufficient bc all dynamics
-        # models have the same auxiliary task model and features
-        # TODO: should the feature model be independent of dynamics?
-        self.dynamics_list[0].auxiliary_task.update_features(obs, last_obs)
+    def auxiliary_loss(self):
         # Get the loss and variance of the feature model
-        aux_loss = torch.mean(
-            self.dynamics_list[0].auxiliary_task.get_loss()
-        )
+        aux_loss = torch.mean(self.auxiliary_task.get_loss())
         # Take variance over steps -> [feature_dim] vars -> average
         # This is the average variance in a feature over time
-        feature_var = torch.mean(
-            torch.var(self.dynamics_list[0].auxiliary_task.features, [0, 1])
-        )
-        feature_var_2 = torch.mean(
-            torch.var(self.dynamics_list[0].auxiliary_task.features, [2])
-        )
+        feature_var = torch.mean(torch.var(self.auxiliary_task.features, [0, 1]))
+        feature_var_2 = torch.mean(torch.var(self.auxiliary_task.features, [2]))
         return aux_loss, {
             "phi/feature_var_ax01": feature_var,
             "phi/feature_var_ax2": feature_var_2,
             "loss/auxiliary_task": aux_loss
         }
 
-    def backprop_loss(self, acs, obs, last_obs):
-        pred_features = []
-        # Get output from all dynamics models (featurewise)
-        # shape=[num_dynamics, num_envs, n_steps_per_seg, feature_dim]
+    def update_auxiliary_task(self, acs, obs, last_obs, return_next_features=True):
+        # Update the auxiliary task
+        self.auxiliary_task.policy.update_features(obs, acs)
```

Comment:
I think we could skip this here if we are not sharing features with the policy. This would save us a forward pass through the policy network. If we don't call policy.update_features we have to remember to pass the current action batch to the auxiliary_task update or simply remove acs altogether from this function since there is no change happening on it (asc==ac)